pkg/networkservice/connectioncontext/ipcontext/vl3/ipam.go

```diff
@@ -34,6 +34,13 @@ type IPAM struct {
 	clientMask       uint8
 }
 
+// NewIPAM creates a new vl3 ipam with specified prefix and excluded prefixes
+func NewIPAM(ctx context.Context, prefix string, excludedPrefixes []string) *IPAM {
```

Comment:
I think we should rework the reset API and not use the context. We can simply return an error to avoid extrapameters. 

```suggestion
func NewIPAM(prefix string, excludedPrefixes...string) *IPAM {
```